ansible/rke2/airgap/roles/ssh_setup/tasks/main.yml

```diff
@@ -0,0 +1,141 @@
+---
+- name: Ensure .ssh directory exists
+  file:
+    path: "~{{ ansible_user }}/.ssh"
+    state: directory
+    mode: '0700'
+    owner: "{{ ansible_user }}"
+    group: "{{ ansible_user }}"
+  become: true
+
+- name: Copy SSH private key with correct filename
+  copy:
+    src: "{{ ssh_private_key_file }}"
+    dest: "~{{ ansible_user }}/.ssh/{{ ssh_private_key_file | basename }}"
+    mode: '0600'
```

Comment:
Private key I believe should be `0400` -- I think there is a restriction in AWS if using more permissive than this.

Reply:
It seems to be working as is. I've always used rw+u access for my private keys without issue. Is there some notification that AWS would give for incorrect permissions, or would it show up in a security scan? I guess what I'm saying is that if your user account is compromised then the attacker could just change the user permissions to whatever they wanted to.